### PR TITLE
Analytics service invocation

### PR DESCRIPTION
**Pull request by sohanirao**

### CODE REVIEW

File: src/runtime/analytics-service.js [hunk 1]
```diff
@@ -80,6 +80,12 @@ export class AnalyticsService {
      * @private @const {!TransactionId}
      */
     this.xid_ = new TransactionId(deps);
+
+    /** @private {?Promise} */
+    this.serviceReady_ = null;
+
+    /** {?Promise} */
+    this.lastAction = null;
```
[the remark this answers is not in the document]
Comment (dvoytenko) [reply src/runtime/analytics-service.js]: Why not? Let's make it private.

File: src/runtime/analytics-service.js [hunk 2]
```diff
@@ -138,14 +146,43 @@ export class AnalyticsService {
...
+   * @private
    */
-  start() {
-    this.doc_.getBody().appendChild(this.getElement());
-    return this.activityPorts_.openIframe(this.iframe_, this.src_,
-        this.args_).then(port => {
-          this.portResolver_(port);
-          return this.setContext_();
+  start_() {
+    return this.init_().then(() => {
```
Comment (dvoytenko) [inline src/runtime/analytics-service.js]: The `start` and `init` and how they cooperate right now is very complicated, but I don't see why that should be the case. Correct me if I'm wrong, but this should be enough?

```
/** @private {!Promise<!ActivityPort>} */
this.serviceReady_ = null;

start_() {
  if (!this.serviceReady_) {
    this.doc_.getBody().appendChild(this.getElement());
    this.serviceReady_ = this.activityPorts_.openIframe(...)
        .then(port => {
            this.setContext_();
            return port.whenReady().then(() => port);
        });
  }
  return this.serviceReady_;
}

logEvent(event) {
  this.stary_().then(port => {
    port.message({...});
  });
}
```

So, the other vars/methods `port_`, `portResolver_`, `portReady_`, etc - all not needed...

File: src/runtime/analytics-service.js [hunk 3]
```diff
@@ -137,15 +142,25 @@ export class AnalyticsService {
   }
 
   /**
-   * @return {!Promise}
+   * @private
```
[the remark this answers is not in the document]
Comment (dvoytenko) [reply src/runtime/analytics-service.js]: Now it should be `{!Promise<!ActivityPort>}`

File: src/runtime/analytics-service.js [hunk 4]
```diff
@@ -80,6 +66,12 @@ export class AnalyticsService {
      * @private @const {!TransactionId}
      */
     this.xid_ = new TransactionId(deps);
+
+    /** @private {?Promise} */
```
Comment (dvoytenko) [inline src/runtime/analytics-service.js]: Now it should be `{?Promise<!ActivityPort>}`

File: src/runtime/analytics-service.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dvoytenko) [reply src/runtime/analytics-service.js]: Ping on this. Keeping it public will cause our obfuscator to not work on this var, but we definitely want to.

File: src/runtime/analytics-service.js (hunk 4 above)
Comment (sohanirao) [reply src/runtime/analytics-service.js]: done

File: src/runtime/analytics-service.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sohanirao) [reply src/runtime/analytics-service.js]: done

**Pull request merged**